app/models/metric/targets.rb

```diff
@@ -32,20 +32,15 @@ def self.capture_cloud_targets(zone, options = {})
     # 1) need to find all the VMs under all the availability zones
     # 2) need to find all the VMs that may not be in an availability zone
     # 3) cloudy clusters?
-    targets = []
+    return [] if options[:exclude_vms]
 
-    includes = {:availability_zones => :tags}
+    includes = {:vms => [{:availability_zones => :tags}, :ext_management_system]}
     MiqPreloader.preload(zone.ems_clouds, includes)
 
-    targets += capture_vm_targets(zone.availability_zones, AvailabilityZone, options)
-
-    # Unlike in the infra world--where every VM must be on a Host--in the cloud
-    #   world (at least in OpenStack) some VMs might not be in an availability
-    #   zone ... in fact, the out-of-the-box setting in nova for the default
-    #   availability zone applied to new VMs is <NONE>.
-    # Track down those cloudy VMs that have no availability zone
-    targets += zone.vms_without_availability_zone unless options[:exclude_vms]
-    targets
+    zone.ems_clouds.flat_map(&:vms).select do |vm|
+      (vm.availability_zone.nil? || vm.availability_zone.perf_capture_enabled?) &&
+        vm.ext_management_system && vm.state == 'on'
+    end
```

Comment:
Ok, @kbrock I don't really know what the code originally did in detail, but capture_vm_targets is a 13 line method and vms_without_availability_zone is a 4 line method.  You've replaced them with 4 lines, which is awesome but it would be helpful to list what each previously did here and how you can replace each part with the changes you've made.